include/api.php

```diff
@@ -6275,15 +6268,17 @@ function api_saved_searches_list($type)
 	$terms = DBA::select('search', ['id', 'term'], ['uid' => local_user()]);
 
 	$result = [];
-	while ($term = $terms->fetch()) {
-		$result[] = [
-			'created_at' => api_date(time()),
-			'id' => intval($term['id']),
-			'id_str' => $term['id'],
-			'name' => $term['term'],
-			'position' => null,
-			'query' => $term['term']
-		];
+	if (DBA::isResult($terms)) {
```

Comment:
This check is superfluous, if `$terms` is invalid or empty `fetch()` will return false anyway.

Reply:
For sure? I got a null-pointer during developing at some point that `$terms` wasn't initialized, that's why I introduced it (because of some forgotten DB-entries ;-) ). But in production, it won't occur so I can remove it again :)

Reply:
See `src/Database/Database.php:863`: https://github.com/friendica/friendica/blob/eefcf99af47013a7837a00910569b2b964e2bf36/src/Database/Database.php#L855-L865

Reply:
Ooooh, I see what you're doing, instead of `$terms->fetch()` it should be `Database::fetch($terms)`.

Reply:
It's DBA::fetch because of the encapsulation. But I see what you mean